sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/utils-service/_tests_/unit/application-event-code-during-enrolment-and-completed-utils.service.applicationEventCodeDuringCompleted.spec.ts

```diff
@@ -0,0 +1,219 @@
+      `COE status is ${COEStatus.completed}, disbursement date is same or after the cutoff date(i.e disbursement date + ${DISBURSEMENT_FILE_GENERATION_ANTICIPATION_DAYS}) ` +
+      `and does not have any ${RestrictionActionType.StopFullTimeDisbursement}.`,
+    async () => {
+      // Arrange and act.
+      const applicationEventCode =
+        await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+          payload,
+          [[RestrictionActionType.StopFullTimeApply]],
+        );
+      // Assert.
+      expect(applicationEventCode).toBe(ApplicationEventCode.COEA);
+    },
+  );
+
+  it(
+    `Should return ${ApplicationEventCode.COER} when the disbursement schedule status is ${DisbursementScheduleStatus.Pending}, ` +
+      `COE status is ${COEStatus.required}.`,
+    async () => {
+      // Arrange and act.
+      const applicationEventCode =
+        await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+          {
+            ...payload,
+            coeStatus: COEStatus.required,
+          },
+        );
+      // Assert.
+      expect(applicationEventCode).toBe(ApplicationEventCode.COER);
+    },
+  );
+
+  it(
+    `Should return ${ApplicationEventCode.COED} when the disbursement schedule status is ${DisbursementScheduleStatus.Pending}, ` +
+      `COE status is ${COEStatus.declined}.`,
+    async () => {
+      // Arrange and act.
+      const applicationEventCode =
+        await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+          {
+            ...payload,
+            coeStatus: COEStatus.declined,
+          },
+        );
+      // Assert.
+      expect(applicationEventCode).toBe(ApplicationEventCode.COED);
+    },
+  );
+
+  it(
+    `Should return ${ApplicationEventCode.DISE} when the disbursement schedule status is ${DisbursementScheduleStatus.Sent}, ` +
+      `and has any full time disbursement feedback errors.`,
+    async () => {
+      // Arrange and act.
+      const applicationEventCode =
+        await applicationEventCodeDuringEnrolmentAndCompletedUtilsService.applicationEventCodeDuringCompleted(
+          {
+            ...payload,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+            disbursementFeedbackErrors: [
+              {
+                errorCode: FULL_TIME_DISBURSEMENT_FEEDBACK_ERRORS[0],
+              },
+            ],
+          } as DisbursementScheduleForApplicationEventCodeDuringCompleted,
+        );
+      // Assert.
+      expect(applicationEventCode).toBe(ApplicationEventCode.DISE);
+    },
+  );
+
+  it(
+    `Should return ${ApplicationEventCode.DISW} when the disbursement schedule status is ${DisbursementScheduleStatus.Sent},` +
+      ` does not have any full time disbursement feedback errors and an award was withheld due to some restriction.`,
```

Comment:
Minor but we can have the white spaces all added to the end of the previous line.